src/uct/tcp/sockcm/sockcm_ep.c

```diff
@@ -14,26 +14,267 @@
         } \
     } while (0)
 
+ucs_status_t uct_sockcm_ep_set_sock_id(uct_sockcm_iface_t *iface, uct_sockcm_ep_t *ep)
+{
+    ucs_status_t status;
+    struct sockaddr *dest_addr = NULL;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+
+    ep->sock_id_ctx = ucs_malloc(sizeof(*ep->sock_id_ctx), "client sock_id_ctx");
+    if (ep->sock_id_ctx == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto out;
+    }
+
+    dest_addr = (struct sockaddr *) &(ep->remote_addr);
+
+    status = ucs_socket_create(dest_addr->sa_family, SOCK_STREAM,
+                               &ep->sock_id_ctx->sock_id);
+    if (status != UCS_OK) {
+        ucs_debug("unable to create client socket for sockcm");
+        goto out_free;
+    }
+
+    ucs_list_add_tail(&iface->used_sock_ids_list, &ep->sock_id_ctx->list);
+    ucs_debug("ep %p, new sock_id %d", ep, ep->sock_id_ctx->sock_id);
+    status = UCS_OK;
+    goto out;
+
+out_free:
+    ucs_free(ep->sock_id_ctx);
+out:
+    UCS_ASYNC_UNBLOCK(iface->super.worker->async);
+    return status;
+}
+
+ucs_status_t uct_sockcm_ep_send_client_info(uct_sockcm_iface_t *iface, uct_sockcm_ep_t *ep)
+{
+    uct_sockcm_conn_param_t conn_param;
+    uct_sockcm_priv_data_hdr_t *hdr;
+    ssize_t transfer_len = 0;
+    ssize_t sent_len = 0;
+
+    memset(&conn_param, 0, sizeof(uct_sockcm_conn_param_t));
+    hdr = &conn_param.hdr;
+
+    /* pack worker address into private data */
+    hdr->length = ep->pack_cb(ep->pack_cb_arg, "DEV_NAME_NULL",
+                              (void*)conn_param.private_data);
+    if (hdr->length < 0) {
+        ucs_error("sockcm client (iface=%p, ep = %p) failed to fill "
+                  "private data. status: %s",
+                  iface, ep, ucs_status_string(hdr->length));
+        return UCS_ERR_IO_ERROR;
+    }
+    ucs_assert(hdr->length + sizeof(int) <= UCT_SOCKCM_PRIV_DATA_LEN);
```

Comment:
You're right. We don't need `sizeof(int)`. In fact the assertion need only check if `hdr->length <= UCT_SOCKCM_PRIV_DATA_LEN` because private_data has space available for just `UCT_SOCKCM_PRIV_DATA_LEN` bytes.